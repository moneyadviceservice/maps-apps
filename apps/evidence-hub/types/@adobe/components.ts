export interface LinkType {
  linkTo: string;
  text: string;
  externalLink?: boolean;
  description: string | null;
}

export type LinkGroup = {
  title: string | null;
  childLinks: LinkType[];
};

export interface Logo {
  image: ImageType;
  altText: string;
}

export interface ImageType {
  _path: string;
  width: number;
  height: number;
  mimeType: string;
}
