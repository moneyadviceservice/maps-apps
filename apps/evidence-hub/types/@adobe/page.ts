import { JsonRichText } from '@maps-react/vendor/utils/RenderRichText';

import { LinkType } from './components';

export interface Tag {
  name: string;
  key: string;
}

export interface DocumentTemplate {
  seoTitle: string;
  seoDescription: string;
  title: string;
  slug: string;
  publishDate: string;
  lastUpdatedDate?: string;
  pageType?: Tag;
  dataType?: Tag[];
  tags?: Tag[];
  breadcrumbs?: LinkType[];
  sections: JsonRichText[];
  contactInformation: JsonRichText;
  clientGroup: Tag[];
  topic: Tag[];
  countryOfDelivery: Tag[];
  links: LinkType[];
  overview?: JsonRichText;
}

export interface TagListItem {
  tagGroup: {
    label: string;
    tags: Tag[];
  };
}
