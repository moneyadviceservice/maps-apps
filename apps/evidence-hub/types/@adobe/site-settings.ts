import { LinkGroup, LinkType, Logo } from './components';

export interface SiteSettings {
  seoTitle: string;
  seoDescription: string;
  headerLogo: Logo;
  mainNavigation: LinkType[];
  footerLinks: LinkGroup[];
}
