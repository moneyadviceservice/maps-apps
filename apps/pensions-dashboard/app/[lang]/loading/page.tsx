import { Metadata, NextPage } from 'next';
import { PensionsDashboardLayout } from '../../../layouts/PensionsDashboardLayout';
import { SITE_TITLE } from '../../../utils/constants';
import { Paragraph } from '@maps-digital/ui/components/Paragraph';
import { Link } from '@maps-digital/ui/components/Link';

const title = 'Pensions found';

export const generateMetadata = async (): Promise<Metadata> => {
  return {
    title: `${title} | ${SITE_TITLE}`,
  };
};

const Page: NextPage = async () => {
  // (1) CDA FIND - 25781
  // > check if data load has at least started, if not complete
  // - use GET not found
  // - or store a cookie when POST has started

  // if NO

  // - create userSessionId UUID
  // - SET cookie
  //   - `userSessionId`
  // - POST to `/api/redirect-details` with
  //   - redirectPurpose = "FIND",
  //   - userSessionId,
  //   - iss = "MHPD-75b68255-444e-4d5f-bbfe-249c26d69963"} (react app env var)
  // - SET cookie, from response
  //   - `codeVerifier`
  // - redirect to CDA emulator
  //   -redirectTargetUrl
  //   - query params from (spread data)
  //   - `redirect_uri` with return URL to React app
  //   - `{redirectTargetUrl}?redirect_uri=http://localhost:4100`

  return (
    <PensionsDashboardLayout title={title}>
      <>
        <Paragraph>Loading...</Paragraph>
        <Link href="/en/overview">Overview</Link>
      </>
    </PensionsDashboardLayout>
  );
};

export default Page;
