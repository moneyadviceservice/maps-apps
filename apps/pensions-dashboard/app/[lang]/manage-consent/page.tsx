import { Metadata, NextPage } from 'next';
import { PensionsDashboardLayout } from '../../../layouts/PensionsDashboardLayout';
import { SITE_TITLE } from '../../../utils/constants';
import { Paragraph } from '@maps-digital/ui/components/Paragraph';

const title = 'Manage Consent';

export const generateMetadata = async (): Promise<Metadata> => {
  return {
    title: `${title} | ${SITE_TITLE}`,
  };
};

const Page: NextPage = async () => {
  return (
    <PensionsDashboardLayout title={title}>
      <Paragraph className="text-center py-24">
        A page to manage consent
      </Paragraph>
    </PensionsDashboardLayout>
  );
};

export default Page;
