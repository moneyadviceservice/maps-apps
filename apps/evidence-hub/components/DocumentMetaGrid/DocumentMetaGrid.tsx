import { twMerge } from 'tailwind-merge';
import { DocumentTemplate } from 'types/@adobe/page';

import { ListElement } from '@maps-react/common/index';

type Props = {
  doc: DocumentTemplate;
};

export const DocumentMetaGrid = ({ doc }: Props) => {
  const columns: { title: string; items: string[] }[] = [
    {
      title: 'Evidence type',
      items: [
        ...(doc.pageType ? [doc.pageType.name || doc.pageType.key] : []),
        ...(doc.dataType && doc.dataType.length > 0
          ? doc.dataType.map((tag) => tag.name || tag.key)
          : []),
      ],
    },
    {
      title: 'Topics',
      items: doc.topic ? doc.topic.map((tag) => tag.name || tag.key) : [],
    },
    {
      title: 'Country',
      items: doc.countryOfDelivery
        ? doc.countryOfDelivery.map((tag) => tag.name || tag.key)
        : [],
    },
    {
      title: 'Population Group',
      items: doc.clientGroup
        ? doc.clientGroup.map((tag) => tag.name || tag.key)
        : [],
    },
  ];

  const visibleColumns = columns.filter((col) => col.items.length > 0);
  const colCount = visibleColumns.length || 1;

  const gridColsClass: Record<number, string> = {
    1: 'sm:grid-cols-1',
    2: 'sm:grid-cols-2',
    3: 'sm:grid-cols-3',
    4: 'sm:grid-cols-4',
    5: 'sm:grid-cols-5',
  };

  return (
    <div
      className={twMerge(
        'grid-cols-1',
        gridColsClass[colCount],
        `grid text-sm text-gray-700 py-2 divide-y-2 sm:divide-y-0 sm:divide-x-2 divide-gray-200`,
      )}
    >
      {visibleColumns.map((col) => (
        <div key={col.title} className={'sm:px-2 pt-2 pb-7'}>
          <p className="mb-2 font-semibold">{col.title}</p>
          <div className="pl-2">
            <ListElement
              items={col.items}
              color="blue"
              variant="unordered"
              className="pl-4"
            />
          </div>
        </div>
      ))}
    </div>
  );
};
