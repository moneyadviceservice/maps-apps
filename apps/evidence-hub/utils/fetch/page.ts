import { DocumentTemplate } from 'types/@adobe/page';

import { aemHeadlessClient } from '@maps-react/utils/aemHeadlessClient';

type PageError = {
  error: boolean;
};

export const fetchPage = async (
  lang: string,
  slug: string,
): Promise<DocumentTemplate | PageError> => {
  try {
    const queryName = `evidence-hub/get-page-${lang};slug=${slug}`;

    const {
      data: {
        pageTemplateList: { items: pageTemplate },
      },
    } = await aemHeadlessClient.runPersistedQuery(queryName);

    if (pageTemplate.length === 0) {
      return {
        error: true,
      } as PageError;
    }

    const result = pageTemplate[0];

    return result;
  } catch (error) {
    console.error('failed to fetch page ', error);
    return {
      error: true,
    } as PageError;
  }
};
