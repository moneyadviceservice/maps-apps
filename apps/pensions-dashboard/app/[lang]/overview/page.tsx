import { Metadata, NextPage } from 'next';
import { twMerge } from 'tailwind-merge';
import { PensionsDashboardLayout } from '../../../layouts/PensionsDashboardLayout';
import { SITE_TITLE, NUMBER_WORDS } from '../../../utils/constants';
import { getPensionsOverview } from '../../../utils/fetch/get-pensions-overview';
import { Button } from '@maps-digital/ui/components/Button';
import { Callout, CalloutVariant } from '@maps-digital/ui/components/Callout';
import { Heading } from '@maps-digital/ui/components/Heading';
import { Icon, IconType } from '@maps-digital/ui/components/Icon';

const title = 'Pensions found';

export const generateMetadata = async (): Promise<Metadata> => {
  return {
    title: `${title} | ${SITE_TITLE}`,
  };
};

const Page: NextPage = async () => {
  const {
    totalPensions,
    incompletePensions,
    confirmedPensions,
    unconfirmedPensions,
  } = await getPensionsOverview();

  const calloutClasses = 'mb-6 p-8';
  const calloutHeadingClasses = 'mb-4';
  const calloutItemClasses = 'flex gap-2 mt-4';
  const calloutCtaClasses = 'mb-12';

  const numberToWords = (num: number) =>
    num > NUMBER_WORDS.length ? num : NUMBER_WORDS[num].en;

  return (
    <PensionsDashboardLayout title={title}>
      <Heading level="h2" className="mb-12">
        We found {numberToWords(totalPensions)} pension
        {totalPensions === 1 ? '' : 's'}
      </Heading>

      <div className="grid grid-cols-3 gap-16">
        <div className="col-span-3 lg:col-span-2">
          {confirmedPensions.length > 0 && (
            <>
              <Callout
                variant={CalloutVariant.POSITIVE}
                className={twMerge(calloutClasses)}
              >
                <Heading level="h6" className={twMerge(calloutHeadingClasses)}>
                  Pensions with complete information
                </Heading>
                <ul>
                  {confirmedPensions.map(
                    ({
                      externalAssetId,
                      pensionType,
                      schemeName,
                      pensionAdministrator: { name },
                    }) => (
                      <li
                        key={externalAssetId}
                        className={twMerge(calloutItemClasses)}
                      >
                        <Icon type={IconType.TICK_GREEN} />
                        {pensionType === 'SP' ? schemeName : name}
                      </li>
                    ),
                  )}
                </ul>
              </Callout>
              <Button
                as="a"
                href="/en/your-pensions"
                className={twMerge(calloutCtaClasses)}
              >
                See pension breakdown
              </Button>
            </>
          )}

          {incompletePensions.length > 0 && (
            <>
              <Callout
                variant={CalloutVariant.WARNING}
                className={twMerge(calloutClasses)}
              >
                <Heading level="h6" className={twMerge(calloutHeadingClasses)}>
                  Pensions waiting for more information from your pension
                  providers
                </Heading>
                <ul>
                  {incompletePensions.map(
                    ({ externalAssetId, pensionAdministrator: { name } }) => (
                      <li
                        key={externalAssetId}
                        className={twMerge(calloutItemClasses)}
                      >
                        <Icon type={IconType.EDIT} />
                        {name}
                      </li>
                    ),
                  )}
                </ul>
              </Callout>
              <Button
                as="a"
                href="/en/pending-pensions"
                className={twMerge(calloutCtaClasses)}
              >
                View pending pensions
              </Button>
            </>
          )}

          {unconfirmedPensions.length > 0 && (
            <>
              <Callout
                variant={CalloutVariant.NEGATIVE}
                className={twMerge(calloutClasses)}
              >
                <Heading level="h6" className={twMerge(calloutHeadingClasses)}>
                  Pensions that are a possible match with your details
                </Heading>
                <ul>
                  {unconfirmedPensions.map(
                    ({ externalAssetId, pensionAdministrator: { name } }) => (
                      <li
                        key={externalAssetId}
                        className={twMerge(calloutItemClasses)}
                      >
                        <Icon
                          type={IconType.WARNING}
                          className="scale-50 m-[-12px]"
                        />
                        {name}
                      </li>
                    ),
                  )}
                </ul>
              </Callout>
              <Button
                as="a"
                href="/en/pensions-that-need-action"
                className={twMerge(calloutCtaClasses)}
              >
                Review pensions that need action
              </Button>
            </>
          )}
        </div>
      </div>
    </PensionsDashboardLayout>
  );
};

export default Page;
