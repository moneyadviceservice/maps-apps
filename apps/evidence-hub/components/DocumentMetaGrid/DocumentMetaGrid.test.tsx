import { DocumentTemplate } from 'types/@adobe/page';
import { render } from '@testing-library/react';

import { DocumentMetaGrid } from './DocumentMetaGrid';

import '@testing-library/jest-dom';

const baseDoc: DocumentTemplate = {
  seoTitle: 'SEO Title',
  seoDescription: 'SEO Description',
  title: 'Test Doc',
  slug: 'test-doc',
  publishDate: '2022-03-15T00:00:00Z',
  sections: [],
  contactInformation: { json: [] },
  topic: [{ key: 'health', name: 'Health' }],
  countryOfDelivery: [{ key: 'us', name: 'USA' }],
  links: [],
  pageType: { key: 'page-report', name: 'Page Report' },
  dataType: [{ key: 'data-report', name: 'Data Report' }],
  clientGroup: [{ key: 'adults', name: 'Adults' }],
  tags: [],
  breadcrumbs: [],
};

describe('DocumentMetaGrid', () => {
  it('renders all four columns when data is present', () => {
    const { container, getByText } = render(<DocumentMetaGrid doc={baseDoc} />);

    expect(getByText('Evidence type')).toBeInTheDocument();
    expect(getByText('Topics')).toBeInTheDocument();
    expect(getByText('Country')).toBeInTheDocument();
    expect(getByText('Population Group')).toBeInTheDocument();

    expect(getByText('Page Report')).toBeInTheDocument();
    expect(getByText('Data Report')).toBeInTheDocument();
    expect(getByText('Health')).toBeInTheDocument();
    expect(getByText('USA')).toBeInTheDocument();
    expect(getByText('Adults')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('renders only visible columns when some fields are missing', () => {
    const partialDoc: DocumentTemplate = {
      ...baseDoc,
      pageType: undefined,
      dataType: undefined,
      topic: [],
      countryOfDelivery: [],
      clientGroup: [],
    };

    const { queryByText } = render(<DocumentMetaGrid doc={partialDoc} />);

    expect(queryByText('Evidence type')).not.toBeInTheDocument();
    expect(queryByText('Topics')).not.toBeInTheDocument();
    expect(queryByText('Country')).not.toBeInTheDocument();
    expect(queryByText('Population Group')).not.toBeInTheDocument();

    expect(queryByText('Page Report')).not.toBeInTheDocument();
    expect(queryByText('Data Report')).not.toBeInTheDocument();
    expect(queryByText('Health')).not.toBeInTheDocument();
    expect(queryByText('USA')).not.toBeInTheDocument();
    expect(queryByText('Adults')).not.toBeInTheDocument();
  });
});
