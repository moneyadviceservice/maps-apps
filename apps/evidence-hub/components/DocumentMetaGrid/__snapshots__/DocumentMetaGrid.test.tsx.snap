// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentMetaGrid renders all four columns when data is present 1`] = `
<div>
  <div
    class="grid-cols-1 sm:grid-cols-4 grid text-sm text-gray-700 py-2 divide-y-2 sm:divide-y-0 sm:divide-x-2 divide-gray-200"
  >
    <div
      class="sm:px-2 pt-2 pb-7"
    >
      <p
        class="mb-2 font-semibold"
      >
        Evidence type
      </p>
      <div
        class="pl-2"
      >
        <div
          class="flex items-center"
          data-testid="list-element"
        >
          <ul
            class="marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug list-disc marker:text-blue-700 pl-4"
            start="-1"
          >
            <li>
              Page Report
            </li>
            <li>
              Data Report
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="sm:px-2 pt-2 pb-7"
    >
      <p
        class="mb-2 font-semibold"
      >
        Topics
      </p>
      <div
        class="pl-2"
      >
        <div
          class="flex items-center"
          data-testid="list-element"
        >
          <ul
            class="marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug list-disc marker:text-blue-700 pl-4"
            start="-1"
          >
            <li>
              Health
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="sm:px-2 pt-2 pb-7"
    >
      <p
        class="mb-2 font-semibold"
      >
        Country
      </p>
      <div
        class="pl-2"
      >
        <div
          class="flex items-center"
          data-testid="list-element"
        >
          <ul
            class="marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug list-disc marker:text-blue-700 pl-4"
            start="-1"
          >
            <li>
              USA
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="sm:px-2 pt-2 pb-7"
    >
      <p
        class="mb-2 font-semibold"
      >
        Population Group
      </p>
      <div
        class="pl-2"
      >
        <div
          class="flex items-center"
          data-testid="list-element"
        >
          <ul
            class="marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug list-disc marker:text-blue-700 pl-4"
            start="-1"
          >
            <li>
              Adults
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
